plugins/rollbar-backend/src/api/RollbarApi.ts

```diff
@@ -0,0 +1,216 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import axios from 'axios';
+import { Logger } from 'winston';
+import camelcaseKeys from 'camelcase-keys';
+import { buildQuery } from '../util';
+import {
+  RollbarItemCount,
+  RollbarItemsResponse,
+  RollbarProject,
+  RollbarProjectAccessToken,
+  RollbarTopActiveItem,
+} from './types';
+
+const baseUrl = 'https://api.rollbar.com/api/1';
+
+const buildUrl = (url: string) => `${baseUrl}${url}`;
+
+export class RollbarApi {
+  private projectMap: ProjectMetadataMap | undefined;
+
+  constructor(
+    private readonly logger: Logger,
+    private readonly accessToken?: string,
+  ) {
+    this.accessToken = accessToken || getRollbarToken(logger);
+  }
+
+  async getAllProjects() {
+    return this.get<RollbarProject[]>('/projects').then(projects =>
+      projects.filter(p => p.name),
+    );
+  }
+
+  async getProject(projectName: string) {
+    return this.getForProject<RollbarProject>(
+      `/project/:projectId`,
+      projectName,
+      false,
+    );
+  }
+
+  async getProjectItems(projectName: string) {
+    return this.getForProject<RollbarItemsResponse>(
+      `/items`,
+      projectName,
+      true,
+    );
+  }
+
+  async getTopActiveItems(
+    projectName: string,
+    options: { hours: number; environment: string } = {
+      hours: 24,
+      environment: 'production',
+    },
+  ) {
+    return this.getForProject<RollbarTopActiveItem[]>(
+      `/reports/top_active_items?${buildQuery(options)}`,
+      projectName,
+    );
+  }
+
+  async getOccuranceCounts(
+    projectName: string,
+    options: { environment: string; item_id?: number } = {
+      environment: 'production',
+    },
+  ) {
+    return this.getForProject<RollbarItemCount[]>(
+      `/reports/occurrence_counts?${buildQuery(options as any)}`,
+      projectName,
+    );
+  }
+
+  async getActivatedCounts(
+    projectName: string,
+    options: { environment: string; item_id?: number } = {
+      environment: 'production',
+    },
+  ) {
+    return this.getForProject<RollbarItemCount[]>(
+      `/reports/activated_counts?${buildQuery(options as any)}`,
+      projectName,
+    );
+  }
+
+  private async getProjectAccessTokens(projectId: number) {
+    return this.get<RollbarProjectAccessToken[]>(
+      `/project/${projectId}/access_tokens`,
+    );
+  }
+
+  private async get<T>(url: string, accessToken?: string) {
+    const fullUrl = buildUrl(url);
+
+    if (this.logger) {
+      this.logger.info(`Calling Rollbar REST API, ${fullUrl}`);
+    }
+
+    return axios
+      .get(fullUrl, getRequestHeaders(accessToken || this.accessToken || ''))
+      .catch(error => {
+        this.logError(error);
+        throw Error(error);
+      })
+      .then(response =>
+        camelcaseKeys<T>(response?.data?.result, { deep: true }),
```

Comment:
Rollbar uses snake_case, so this ensures responses are camelCased to follow JS/TS conventions.

Reply:
I'm a bit indifferent, but I do like to try and respect the local culture as far as using `camelCase` for types in JS/TS. I'd like to extend the backend eventually to support aggregate & historical views, so I can see the backend diverging a bit from the official Rollbar API. If that's the case, then manual mapping will be required anyway.

Reply:
Yep, let's go with it for now anyway